Reproduce dots in quted names

keulinho opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | bug
| Fixed issues | #7030

#### Summary

Provide test for dots in quoted idx names

keulinho committed:
Reproduce dots in quted names

keulinho committed:
Fix test

keulinho committed:
Fix tests

keulinho commented:
The underlying issue was fixed here: https://github.com/doctrine/dbal/commit/92dea81252f05641b07b5be2eb5dd8d5fb73b5eb

Not sure if it makes sense to add the test case just to not accidentally break the behaviour again

anyway I don't know why setting the FK on DB2 and Oracle DBs fails, happy for any insights how to fix

morozov commented on tests/Functional/Schema/SchemaManagerTest.php:
```diff
@@ -275,4 +275,63 @@ public function testIntrospectTableWithInvalidName(): void
+    public function testIntrospectTableWithDotInIndexNames(): void
+    {
+        $table = Table::editor()
+            ->setQuotedName('test_user')
```
> I don't know why setting the FK on DB2 and Oracle DBs fails, happy for any insights how to fix

You declare the table name here as quoted but reference it in the other table as unquoted: https://github.com/doctrine/dbal/blob/d7ff279deea6c631047c83007b0c6db82cf194b6/tests/Functional/Schema/SchemaManagerTest.php#L318

On Oracle and Db2, `"test_user"` (quoted) remains `test_user`, while  `test_user` (unquoted) becomes `TEST_USER`. Hence the error message:

> "DB2INST1.TEST_USER" is an undefined name

Unless you want to retain the original case of the name on all platforms, you should use unquoted names.

morozov committed:
Rebase and fix tests

morozov committed:
Assert that the name is not null at runtime, not statically

Otherwise, an extra "not null" assertion is required before comparing
the value returned by OptionallyNamedObject::getObjectName() with the
expected one.
